### ISSUE TITLE
"File paths are case-sensitive on deployment server" error when deploying to shinyapps.io

### ISSUE THREAD

**Issue by rbcavanaugh** [rstudio/rstudio#11720]
### System details

    RStudio Edition : Desktop
    RStudio Version : 2022.7.1.554
    OS Version      : aarch64-apple-darwin20 (64-bit) macOS Monterey 12.5
    R Version       : 4.2.1 (2022-06-23)

### Steps to reproduce the problem

When deploying shiny app to shinyapps.io, an error for review comes up. It appears to arise from the following line:

```
tidyr::separate(value, into = c("description", "value"), sep = ":|=", extra = "merge")
```

Including this line in an R script for an app deployed to shinyapps.io reproduces the issue for me. As best I can tell, its the inclusion of the word "description" in the into argument. removing description appears to fix the issue for me.

### Describe the problem in detail
 
<img width="764" alt="image" src="https://user-images.githubusercontent.com/46727239/183492364-2db11935-e9c3-42a5-91cc-9b7acad62bf1.png">

<img width="622" alt="image" src="https://user-images.githubusercontent.com/46727239/183491608-1a833dd3-2112-41f5-98b5-30b6402c4aa3.png">

### Describe the behavior you expected

As far as I can tell, this is not an error. Apologies if I'm mistaken or have missed a previous report of this issue.

- [x] I have read the guide for [submitting good bug reports](https://github.com/rstudio/rstudio/wiki/Writing-Good-Bug-Reports).
- [x] I have installed the latest version of RStudio, and confirmed that the issue still persists.
- [x] If I am reporting an RStudio crash, I have included a [diagnostics report](https://support.rstudio.com/hc/en-us/articles/200321257-Running-a-Diagnostics-Report).
- [x] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

**Comment by jmcphers** [comment]
Very likely you have a file named `DESCRIPTION`? If so, does omitting that file make the warning go away?

This is really an issue with the rsconnect package, which has a linter in it that looks for quoted strings that refer to files on disk (RStudio is just showing you the warnings generated by that package). You might open it over there instead: https://github.com/rstudio/rsconnect

**Comment by rbcavanaugh** [comment]
Yup the app is in an R package and changing that filename does resolve the issue. thanks!

**Comment by kevinushey** [comment]
> Yup the app is in an R package and changing that filename does resolve the issue. thanks!

For what it's worth, this is a deficiency in the `rsconnect` linter and you shouldn't need to rename the `DESCRIPTION` file to work around this. I'll move this issue to the `rsconnect` repository.
